src/backends/tensorflow/graphdef_backend_factory.cc

```diff
@@ -69,10 +69,18 @@ GraphDefBackendFactory::CreateBackend(
         std::make_tuple(graphdef_path));
   }
 
+  // Initialize virtual device counter
```

Comment:
I forgot that there is both graphdef and savedmodel factories. You don't want 2 vgpu id maps, just one that is shared. So you should make a singleton utility class that is shared that holds the gpu_device -> vgpu_device logic.